File: Logic/Java/EvaluateExpression.java
public class EvaluateExpression {
    public static int evaluate(String expr) {
        int n = (expr.length() + 1) / 2;  // Number of digits
        String digits = expr.substring(0, n);
        String operators = expr.substring(n);

        int result = digits.charAt(0) - '0';  // Convert char to int

        for (int i = 1; i < n; i++) {
            int num = digits.charAt(i) - '0';  // Convert char to int
            char op = operators.charAt(i - 1);

            switch (op) {
                case '+' -> result += num;
                case '-' -> result -= num;
                case '*' -> result *= num;
                case '/' -> result /= num;
            }
        }

        return result;
    }

    public static void main(String[] args) {
        String expr = "12345*+-+";
        int result = evaluate(expr);
        System.out.println(result);  // Output: 6
    }
}

File: Logic/Python/evaluate_expression.py
def evaluate_expression(expr):
    n = (len(expr) + 1) // 2  # number of digits
    digits = expr[:n]
    operators = expr[n:]

    result = int(digits[0])

    for i in range(1, n):
        num = int(digits[i])
        op = operators[i - 1]

        if op == '+':
            result += num
        elif op == '-':
            result -= num
        elif op == '*':
            result *= num
        elif op == '/':
            result = int(result / num)

    return result

# Test
print(evaluate_expression("12345*+-+"))  # Output: 6
